libyul/backends/evm/NoOutputAssembly.cpp

```diff
@@ -207,9 +208,36 @@ void NoOutputAssembly::appendReturnContract(ContainerID)
 NoOutputEVMDialect::NoOutputEVMDialect(EVMDialect const& _copyFrom):
 	EVMDialect(_copyFrom.evmVersion(), _copyFrom.eofVersion(), _copyFrom.providesObjectAccess())
 {
-	for (auto& fun: m_functions)
-		if (fun)
-			modifyBuiltinToNoOutput(*fun);
+	// save the modified functions here - note that this only has to be done once because we modify all of
+	// them in one go, later just reference pointers to this static vector
+	static std::vector<BuiltinFunctionForEVM> const noOutputBuiltins = []
+	{
+		std::vector<BuiltinFunctionForEVM> modifiedBuiltins;
+		modifiedBuiltins.reserve(allBuiltins().functions().size());
+
+		for (auto const& [_, builtin]: allBuiltins().functions())
+		{
+			auto noOutputFunction = builtin;
+			modifyBuiltinToNoOutput(noOutputFunction);
+			modifiedBuiltins.push_back(std::move(noOutputFunction));
+		}
+
+		return modifiedBuiltins;
+	}();
```

Comment:
I'm not a fan of this pattern. The `()` at the end just stand out enough. And it's not just me: [Improving Readability of IIFE ](https://www.cppstories.com/2016/11/iife-for-complex-initialization/#improving-readability-of-iife).

This two especially would be perfectly fine as normal, static functions with well defined inputs and outputs and descriptive names.

Reply:
Conversely you'd probably be able to find people and evidence who like the pattern. I do, for example. It makes it very easy for me to see that a variable is assigned the result of an expression and the implementation of the expression is right below it. Matter of preference and what your brain can easily digest when reading it, as it is the case with many things :)

Here especially I think the function is quite localized and does not necessarily warrant a 'full' static function, the IIFE improves readability and also clarifies scoping for definition of `m_functions`. The second one is purposefully not static as it depends on the input dialect.

I have modified it so that (I hope) it is more to your liking but I want to note that I think it rather hurts readability. For me at least.